boards/arduino-mega2560/board.c

```diff
@@ -82,12 +82,14 @@ void SystemInit(void)
 
 static int uart_putchar(char c, FILE *stream)
 {
+    (void) stream;
     uart_stdio_write(&c, 1);
     return 0;
 }
 
 int uart_getchar(FILE *stream)
 {
+    (void) stream;
     char c;
     uart_stdio_read(&c, 1);
     return (int)c;
```

Comment:
Kinda unrelated to this PR, but the cast is wrong. It should be `(unsigned char) c` or it will break for UTF-8 or binary input.